lib/src/main/java/com/segment/analytics/kotlin/destinations/amplitude/AmplitudeSession.kt

```diff
@@ -27,14 +33,25 @@ class AmplitudeSession (private val sessionTimeoutMs : Long = 300000) : EventPlu
             return event
         }
 
+        startNewSessionIfNecessary()
+
         val modified = super.execute(event)
         analytics.log(
-            message = "Running ${event.type} payload through AmplitudeSession",
-            kind = LogKind.DEBUG
+            message = "Running ${event.type} payload through AmplitudeSession"
         )
-        lastEventFiredTime = java.lang.System.currentTimeMillis()
 
-        return modified?.putIntegrations(key, mapOf("session_id" to sessionID))
+        if (event is TrackEvent) {
+            if(event.event == ampSessionStartEvent) {
+                // Update session ID for all events after this in the queue
+                eventSessionId = sessionId
+                analytics.log(message = "NewSession = $eventSessionId")
+            }
+            if (event.event == ampSessionEndEvent) {
```

Comment:
maybe add `else` to avoid running to if check twice?
```suggestion
            else if (event.event == ampSessionEndEvent) {
```